tests/conftest.py

```diff
@@ -0,0 +1,113 @@
+"""Reusable test fixtures for ``jupyter_server_proxy``."""
+import os
+import shutil
+import socket
+import sys
+import time
+from pathlib import Path
+from subprocess import Popen
+from typing import Any, Generator, Tuple
+from urllib.error import URLError
+from urllib.request import urlopen
+from uuid import uuid4
+
+from pytest import fixture
+
+HERE = Path(__file__).parent
+RESOURCES = HERE / "resources"
+
+
+@fixture
+def a_token() -> str:
+    """Get a random UUID to use for a token."""
+    return str(uuid4())
+
+
+@fixture
+def an_unused_port() -> int:
+    """Get a random unused port."""
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(("127.0.0.1", 0))
+    s.listen(1)
+    port = s.getsockname()[1]
+    s.close()
+    return port
+
+
+@fixture(params=["notebook", "lab"])
+def a_server_cmd(request: Any) -> str:
+    """Get a viable name for a command."""
+    return request.param
+
+
+@fixture
+def a_server(
+    a_server_cmd: str,
+    tmp_path: Path,
+    an_unused_port: int,
+    a_token: str,
+) -> Generator[str, None, None]:
+    """Get a running server."""
+    # get a copy of the resources
+    tests = tmp_path / "tests"
+    tests.mkdir()
+    shutil.copytree(RESOURCES, tests / "resources")
+    args = [
+        sys.executable,
+        "-m",
+        "jupyter",
+        a_server_cmd,
+        f"--port={an_unused_port}",
+        "--no-browser",
+        "--config=./tests/resources/jupyter_server_config.py",
```

Comment:
E #

Reply:
E ?

Reply:
This was a quick review from the mobile phone while residing in my pocket! Maybe not a blocker though?
I'll put the mobile phone back in the pocket later and hope that it gets marked as resolved!